geopandas/tests/test_plotting.py

```diff
@@ -552,6 +566,37 @@ def test_plot(self):
         self.df.plot()
 
 
+class TestGeometryCollectionPlotting:
+    def setup_method(self):
+        coll1 = GeometryCollection(
+            [
+                Polygon([(1, 0), (2, 0), (2, 1)]),
+                LineString([(0.5, 0.5), (1, 1), (1, 0.5), (1.5, 1)]),
+            ]
+        )
+        coll2 = GeometryCollection(
+            [Point(0.75, 0.25), Polygon([(2, 2), (3, 2), (2, 3)])]
+        )
+
+        self.series = GeoSeries([coll1, coll2])
+        self.df = GeoDataFrame({"geometry": self.series, "values": [1, 2]})
+
+    def test_colors(self):
+        # default uniform color
+        ax = self.series.plot()
+        _check_colors(1, ax.collections[0].get_facecolors(), [MPL_DFT_COLOR])
+        _check_colors(1, ax.collections[1].get_edgecolors(), [MPL_DFT_COLOR])
+        _check_colors(2, ax.collections[2].get_facecolors(), [MPL_DFT_COLOR])
+
+    def test_values(self):
+        ax = self.df.plot("values")
+        cmap = plt.get_cmap()
+        exp_colors = cmap(np.arange(2) / 1)
+        _check_colors(1, ax.collections[0].get_facecolors(), exp_colors)
+        _check_colors(1, ax.collections[1].get_facecolors(), [exp_colors[0]])
+        _check_colors(2, ax.collections[2].get_facecolors(), [exp_colors[1]])
```

Comment:
can you add some comments here about which collection is what and therefore why those exp_colors are expected?